android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -236,6 +235,21 @@ fun setLogHandler(callback: (logDetails: Map<String, String>) -> Unit) {
     Purchases.logHandler = LogHandlerWithMapping(callback)
 }
 
+/**
+ * Sets a log handler and forwards all logs to completion function. Accepts an OnResult so it can be used from
+ * SDKs that don't have Kotlin configured, since they would error because Kotlin's Function1 would not be found.
+ * Function1 is what Kotlin lambdas are converted to. It has a different name than setLogHandler because naming it
+ * the same also gives errors due of missing Function1.
+ *
+ * @param onResult Gets a map with two keys, a `logLevel` which  is one of the ``LogLevel`` name uppercased,
+ * and a `message`, with the log message. The onError of OnResult will never be called.
+ */
+fun setLogHandlerWithOnResult(onResult: OnResult) {
```

Comment:
Open to change the name of the function and also open to create a new functional interface instead of using `OnResult`, although I think since this is internal we can just reuse `OnResult`